keras/applications/regnet.py

```diff
@@ -0,0 +1,1576 @@
+                               kernel_initializer=initializers.HeNormal(),
+                               name=name + "_conv_3x3")
+      
+
+    else:
+      skip = inputs
+      conv_3x3 = layers.Conv2D(filters_out, (3, 3),
+                               use_bias=False,
+                               groups=groups,
+                               padding="same",
+                               kernel_initializer=initializers.HeNormal(),
+                               name=name + "_conv_3x3")
+
+    # Build block
+    # conv_1x1_1
+    x = layers.Conv2D(filters_out, (1, 1),
+                      use_bias=False,
+                      kernel_initializer=initializers.HeNormal(),
+                      name=name + "_conv_1x1_1")(inputs)
+    x = layers.BatchNormalization(momentum=0.9,
+                                  epsilon=1e-5,
+                                  name=name + "_conv_1x1_1_bn")(x)
+    x = layers.ReLU(name=name + "_conv_1x1_1_relu")(x)
+
+    # # conv_3x3
+    x = conv_3x3(x)
+    x = layers.BatchNormalization(momentum=0.9,
+                                  epsilon=1e-5,
+                                  name=name + "_conv_3x3_bn")(x)
+    x = layers.ReLU(name=name + "_conv_3x3_relu")(x)
+
+    # Squeeze-Excitation block (https://arxiv.org/abs/1709.01507)
+    x = SqueezeAndExciteBlock(filters_out, se_filters, name=name)(x)
+
+    # conv_1x1_2
+    x = layers.Conv2D(filters_out, (1, 1),
+                      use_bias=False,
+                      kernel_initializer=initializers.HeNormal(),
+                      name=name + "_conv_1x1_2")(x)
+    x = layers.BatchNormalization(momentum=0.9,
+                                  epsilon=1e-5,
+                                  name=name + "_conv_1x1_2_bn")(x)
+
+    x = layers.ReLU(name=name + "_exit_relu")(x + skip)
+
+    return x
+
+  return apply
+
+
+def ZBlock(filters_in,
+           filters_out,
+           group_width,
+           stride=1,
+           squeeze_excite_ratio=0.25,
+           bottleneck_ratio=0.25,
+           name=None):
+  """
+  Implementation of Z block
+  Reference: [Fast and Accurate Model Scaling](https://arxiv.org/abs/2103.06877).
+  
+  Args:
+    filters_in: filters in the input tensor
+    filters_out: filters in the output tensor
+    group_width: group width
+    stride: stride
+    squeeze_excite_ratio: expansion ration for Squeeze and Excite block
+    bottleneck_ratio: inverted bottleneck ratio 
+    name: name prefix
+
+  Return:
+    Output tensor of the block 
+  """
+  if name is None:
+    name = str(backend.get_uid("zblock"))
+
+  def apply(inputs):
+    if filters_in != filters_out and stride == 1:
+      raise ValueError(
+          f"""Input filters({filters_in}) and output filters({filters_out}) are not equal for stride
+                          {stride}. Input and output filters must be equal for stride={stride}."""
+      )
+
+    groups = filters_out // group_width
+    se_filters = int(filters_out * squeeze_excite_ratio)
```

Comment:
Following the logic of ebea266af09deb81075a96483e5fb20c560ea10d should this be `filters_in` as well?
```suggestion
    se_filters = int(filters_in * squeeze_excite_ratio)
```